### PR TITLE
Fix PackageMetadata validation error with extra provided field

### PR DESCRIPTION
**Pull request by aganders3**
The `PackageMetadata` class is configured to ignore extra fields but instantiation can still fail (raises a `KeyError`) when actually provided an extra field due to this custom root validator. Based on the config this does not seem like the intended behavior.

```
class PackageMetadata(BaseModel):
    ....
    class Config:
        extra = Extra.ignore
    ...
    @root_validator(pre=True)
    def _validate_root(cls, values):
        if "metadata_version" not in values:
            fields = cls.__fields__
            mins = {fields[n].field_info.extra.get("min_ver", "1.0") for n in values}
            values["metadata_version"] = str(max(float(x) for x in mins))
        return values
```

This PR just changes that validator skip over any values not in the list of expected fields during this step.

### CODE REVIEW

**Review by nclack — APPROVED** [review]
Thanks! 

You can remove the pyside dependency in the ci. It'll cause a test to fail, but that should get fixed in #255.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ jobs:
           python -m pip install -e ./napari-from-github
           python -m pip install -e .[json]
           # bare minimum required to test napari/plugins
-          python -m pip install pytest scikit-image[data] zarr xarray hypothesis matplotlib
+          python -m pip install pytest scikit-image[data] zarr xarray hypothesis matplotlib pyside6
```
[the remark this answers is not in the document]
Comment (nclack) [reply .github/workflows/ci.yml:68]: to clarify, the `CI/napari tests` check should fail, but that's ok.

[the remark this answers is not in the document]
Comment (aganders3) [reply .github/workflows/ci.yml]: Done - thanks!

**Review by goanpeca — APPROVED** [review]

File: tests/test_package_meta.py [hunk 2]
```diff
@@ -17,3 +17,24 @@ def test_package_metadata_version():
...
 def test_hashable():
     hash(PackageMetadata(name="test", version="1.0"))
+
+
+def test_package_metadata_extra_field():
+    pkg = {
+        "name": "test",
+        "version": "1.0",
+        "maintainer": "bob",
+        "extra_field_that_is_definitely_not_in_the_model": False,
```
Comment (goanpeca) [inline tests/test_package_meta.py:27]: Love this key name 😄

File: tests/test_package_meta.py [hunk 3]
```diff
@@ -17,3 +17,24 @@ def test_package_metadata_version():
...
+    }
+
+    try:
+        p = PackageMetadata(**pkg)
```
Comment (goanpeca) [inline tests/test_package_meta.py:31]: @aganders3 I am confused, so with the fix it should not fail with extra fields? So why is the test checking that it fails. Or am I missing something 🤔

Comment (aganders3) [reply tests/test_package_meta.py:31]: You're right - the try/except is a little ugly but is basically rewriting whatever error would come up into an `AssertionError` for consistency with pytest expectations.

Then after parsing succeeds the other assertions check known fields are present/correct and the extra field is ignored/dropped. Does that make sense?

Comment (goanpeca) [reply tests/test_package_meta.py:31]: Yep :)

**Pull request merged**